R-package/R/xgb.Booster.R

```diff
@@ -375,23 +395,78 @@ predict.xgb.Booster <- function(object, newdata, missing = NA, outputmargin = FA
     type = box(as.integer(0))
   )
 
+  if (length(list(...))) {
+    extra_args <- list(...)
+    extra_argnames <- names(extra_args)
+    is_default_type <- head(type, 1) == "link"
+
+    old_args_for_type <- list(
+      "outputmargin" = "margin",
+      "predleaf" = "leaf",
+      "predcontrib" = "contrib",
+      "predinteraction" = "interaction"
+    )
+
+    for (arg in names(old_args_for_type)) {
+      if (arg %in% extra_argnames) {
+        warning(
```

Comment:
My only concern with `warning` is that it might break CRAN checks of reverse dependencies. We need to talk to their maintainers before pushing this change to CRAN.

Reply:
It most definitly will. Good thing is, next version is going to be a 2.0, which signals a major release and thus potentially breaking changes.

Reply:
In concept for general development this is true, while for CRAN it still regardlessly breaks reverse dependencies and got Ripleyed.

Reply:
Changing the interface of `predict()` will definitively cause chaos, not only on CRAN.